Array multiplication example code needs to be updated everytime kompute have a build issue.

https://github.com/KomputeProject/kompute/blob/637bfb1bc044d5e9a639a5ef975ae4ce4b82a71c/examples/array_multiplication/CMakeLists.txt#L29

As it stands, array_multiplication fetches a hash on github. This example code will need continuous maintenance or else the fails to compile whenever somebody tries to build the example and a build issue is fixed upstream.

Steps to reproduce
1. git clone https://github.com/KomputeProject/kompute.git
2. cd examples/array_multiplication
3. cmake -Bbuild
4. cd build
5. make

```
$make
[ 13%] Built target fmt
[ 22%] Built target kp_shader
[ 27%] Building CXX object _deps/kompute-build/src/logger/CMakeFiles/kp_logger.dir/Logger.cpp.o
[ 31%] Linking CXX static library libkp_logger.a
[ 31%] Built target kp_logger
[ 36%] Building CXX object _deps/kompute-build/src/CMakeFiles/kompute.dir/Algorithm.cpp.o
[ 40%] Building CXX object _deps/kompute-build/src/CMakeFiles/kompute.dir/Manager.cpp.o
[ 45%] Building CXX object _deps/kompute-build/src/CMakeFiles/kompute.dir/OpAlgoDispatch.cpp.o
[ 50%] Building CXX object _deps/kompute-build/src/CMakeFiles/kompute.dir/OpMemoryBarrier.cpp.o
[ 54%] Building CXX object _deps/kompute-build/src/CMakeFiles/kompute.dir/OpTensorCopy.cpp.o
In file included from /home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/operations/OpTensorCopy.hpp:6,
                 from /home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/OpTensorCopy.cpp:3:
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:55:27: error: expected ‘)’ before ‘<’ token
   55 |     Tensor(std::shared_ptr<vk::PhysicalDevice> physicalDevice,
      |           ~               ^
      |                           )
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:108:30: error: ‘std::shared_ptr’ has not been declared
  108 |                         std::shared_ptr<Tensor> copyFromTensor);
      |                              ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:108:40: error: expected ‘,’ or ‘...’ before ‘<’ token
  108 |                         std::shared_ptr<Tensor> copyFromTensor);
      |                                        ^
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:256:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
  256 |     std::shared_ptr<vk::PhysicalDevice> mPhysicalDevice;
      |          ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:6:1: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
    5 | #include "logger/Logger.hpp"
  +++ |+#include <memory>
    6 | #include <string>
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:257:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
  257 |     std::shared_ptr<vk::Device> mDevice;
      |          ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:257:5: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
  257 |     std::shared_ptr<vk::Device> mDevice;
      |     ^~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:260:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
  260 |     std::shared_ptr<vk::Buffer> mPrimaryBuffer;
      |          ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:260:5: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
  260 |     std::shared_ptr<vk::Buffer> mPrimaryBuffer;
      |     ^~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:262:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
  262 |     std::shared_ptr<vk::Buffer> mStagingBuffer;
      |          ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:262:5: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
  262 |     std::shared_ptr<vk::Buffer> mStagingBuffer;
      |     ^~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:264:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
  264 |     std::shared_ptr<vk::DeviceMemory> mPrimaryMemory;
      |          ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:264:5: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
  264 |     std::shared_ptr<vk::DeviceMemory> mPrimaryMemory;
      |     ^~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:266:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
  266 |     std::shared_ptr<vk::DeviceMemory> mStagingMemory;
      |          ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:266:5: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
  266 |     std::shared_ptr<vk::DeviceMemory> mStagingMemory;
      |     ^~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:270:28: error: ‘std::shared_ptr’ has not been declared
  270 |     void createBuffer(std::shared_ptr<vk::Buffer> buffer,
      |                            ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:270:38: error: expected ‘,’ or ‘...’ before ‘<’ token
  270 |     void createBuffer(std::shared_ptr<vk::Buffer> buffer,
      |                                      ^
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:272:34: error: ‘std::shared_ptr’ has not been declared
  272 |     void allocateBindMemory(std::shared_ptr<vk::Buffer> buffer,
      |                                  ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:272:44: error: expected ‘,’ or ‘...’ before ‘<’ token
  272 |     void allocateBindMemory(std::shared_ptr<vk::Buffer> buffer,
      |                                            ^
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:276:32: error: ‘std::shared_ptr’ has not been declared
  276 |                           std::shared_ptr<vk::Buffer> bufferFrom,
      |                                ^~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:276:42: error: expected ‘,’ or ‘...’ before ‘<’ token
  276 |                           std::shared_ptr<vk::Buffer> bufferFrom,
      |                                          ^
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:302:28: error: expected ‘)’ before ‘<’ token
  302 |     TensorT(std::shared_ptr<vk::PhysicalDevice> physicalDevice,
      |            ~               ^
      |                            )
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/OpTensorCopy.cpp: In member function ‘virtual void kp::OpTensorCopy::record(const vk::CommandBuffer&)’:
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/OpTensorCopy.cpp:49:42: error: cannot convert ‘__gnu_cxx::__alloc_traits<std::allocator<std::shared_ptr<kp::Tensor> >, std::shared_ptr<kp::Tensor> >::value_type’ {aka ‘std::shared_ptr<kp::Tensor>’} to ‘int’
   49 |         this->mTensors[i]->recordCopyFrom(commandBuffer, this->mTensors[0]);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/doof/gitrepo/kompute/examples/array_multiplication/build/_deps/kompute-src/src/include/kompute/Tensor.hpp:108:25: note:   initializing argument 2 of ‘void kp::Tensor::recordCopyFrom(const vk::CommandBuffer&, int)’
  108 |                         std::shared_ptr<Tensor> copyFromTensor);
      |                         ^~~~~~~~~~~~~~~
make[2]: *** [_deps/kompute-build/src/CMakeFiles/kompute.dir/build.make:132: _deps/kompute-build/src/CMakeFiles/kompute.dir/OpTensorCopy.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:263: _deps/kompute-build/src/CMakeFiles/kompute.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
````

As you can see here, we recently fix a gcc12 error. The example code does not build because cmake fetches an old unpatch version of kompute since hashes are hard coded.

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -32,8 +32,11 @@ jobs:
     - name: Run tests
       run: |
         # Check that the expected output is printed
-        TEST_STDOUT=./examples/array_multiplication/build/src/kompute_array_mult
+        TEST_STDOUT=$(./examples/array_multiplication/build/src/kompute_array_mult)
+        echo "$TEST_STDOUT"
         TEST_CODE=$?
+        echo "$TEST_CODE"
         TEST_MATCH= echo $TEST_STDOUT | grep -q "Output: {  0  4  12  }"
+        echo "$TEST_MATCH"
         exit $(($TEST_CODE + $TEST_MATCH))
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -20,8 +20,8 @@ jobs:
     - name: "[Release g++] Build & Test"
       uses: KomputeProject/action-cmake-build@master
       with:
-        build-dir: ${{github.workspace}}/examples/array_multiplication/build
-        source-dir: ${{github.workspace}}/examples/array_multiplication
+        build-dir: examples/array_multiplication/build
+        source-dir: examples/array_multiplication
         cc: gcc
         cxx: g++
         build-type: Debug
@@ -32,5 +32,5 @@ jobs:
     - name: Run tests
       run: |
         # Check that the expected output is printed
-        ${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult
+        ./examples/array_multiplication/build/src/kompute_array_mult
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -55,4 +55,4 @@ jobs:
         configure-options: -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON KOMPUTE_OPT_FROM_SOURCE=ON
         build-options: --parallel # Given we don't build too many resources we can leverage parallel
     - name: Run tests
-      run: ./examples/logistic_regression/build/src/kompute_array_mult
+      run: ./examples/logistic_regression/build/src/kompute_logistic_regression

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -32,7 +32,5 @@ jobs:
     - name: Run tests
       run: |
         # Check that the expected output is printed
-        ${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult | \
-          tee $(tty) | grep -q "Output: {  0  4  12  }"
-
+        ${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -20,8 +20,8 @@ jobs:
     - name: "[Release g++] Build & Test"
       uses: KomputeProject/action-cmake-build@master
       with:
-        build-dir: examples/array_multiplication/build
-        source-dir: examples/array_multiplication
+        build-dir: ${{github.workspace}}/examples/array_multiplication/build
+        source-dir: ${{github.workspace}}/examples/array_multiplication
         cc: gcc
         cxx: g++
         build-type: Debug

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -32,5 +32,6 @@ jobs:
     - name: Run tests
       run: |
         # Check that the expected output is printed
-        ./examples/array_multiplication/build/src/kompute_array_mult
+        ./examples/array_multiplication/build/src/kompute_array_mult | \
+          tee $(tty) | grep -q "Output: {  0  4  12  }"
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -31,3 +31,28 @@ jobs:
         build-options: --parallel # Given we don't build too many resources we can leverage parallel
     - name: Run tests
       run: ./examples/array_multiplication/build/src/kompute_array_mult
+
+  logistc-regression-example:
+    runs-on: ubuntu-latest
+    container: axsauze/kompute-builder:0.4
+    env:
+      VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        submodules: false
+    - name: "[Release g++] Build & Test"
+      uses: KomputeProject/action-cmake-build@master
+      with:
+        build-dir: ${{github.workspace}}/examples/logistic_regression/build
+        source-dir: ${{github.workspace}}/examples/logistic_regression
+        cc: gcc
+        cxx: g++
+        build-type: Debug
+        run-test: false
+        ctest-options: -V
+        configure-options: -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON KOMPUTE_OPT_FROM_SOURCE=ON
+        build-options: --parallel # Given we don't build too many resources we can leverage parallel
+    - name: Run tests
+      run: ./examples/logistic_regression/build/src/kompute_array_mult

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -33,6 +33,6 @@ jobs:
       run: |
         # Check that the expected output is printed
         OUTPUT=${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult | \
-          tee /dev/stdout | grep -q "Output: {  0  4  12  }"
+          tee $(tty) | grep -q "Output: {  0  4  12  }"
 
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -1,0 +1,38 @@
+name: C++ Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  array-multiplication-example:
+    runs-on: ubuntu-latest
+    container: axsauze/kompute-builder:0.4
+    env:
+      VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        submodules: false
+    - name: "[Release g++] Build & Test"
+      uses: KomputeProject/action-cmake-build@master
+      with:
+        build-dir: ${{github.workspace}}/examples/array_multiplication/build
+        source-dir: ${{github.workspace}}/examples/array_multiplication
+        cc: gcc
+        cxx: g++
+        build-type: Debug
+        run-test: false
+        ctest-options: -V
+        configure-options: -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON KOMPUTE_OPT_FROM_SOURCE=ON
+        build-options: --parallel # Given we don't build too many resources we can leverage parallel
+    - name: Run tests
+      run: |
+        # Check that the expected output is printed
+        OUTPUT=${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult | \
+          tee /dev/stdout | grep -q "Output: {  0  4  12  }"
+
+

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -35,8 +35,9 @@ jobs:
         TEST_STDOUT=$(./examples/array_multiplication/build/src/kompute_array_mult)
         echo "$TEST_STDOUT"
         TEST_CODE=$?
-        echo "$TEST_CODE"
-        TEST_MATCH= echo $TEST_STDOUT | grep -q "Output: {  0  4  12  }"
-        echo "$TEST_MATCH"
+        echo "Exit value for test run: $TEST_CODE"
+        echo $TEST_STDOUT | grep -q "Output: {  0  4  12  }"
+        TEST_MATCH=$?
+        echo "Exit value for text match: $TEST_MATCH"
         exit $(($TEST_CODE + $TEST_MATCH))
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -30,14 +30,4 @@ jobs:
         configure-options: -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON KOMPUTE_OPT_FROM_SOURCE=ON
         build-options: --parallel # Given we don't build too many resources we can leverage parallel
     - name: Run tests
-      run: |
-        # Check that the expected output is printed
-        TEST_STDOUT=$(./examples/array_multiplication/build/src/kompute_array_mult)
-        echo "$TEST_STDOUT"
-        TEST_CODE=$?
-        echo "Exit value for test run: $TEST_CODE"
-        echo $TEST_STDOUT | grep -q "Output: {  0  4  12  }"
-        TEST_MATCH=$?
-        echo "Exit value for text match: $TEST_MATCH"
-        exit $(($TEST_CODE + $TEST_MATCH))
-
+      run: ./examples/array_multiplication/build/src/kompute_array_mult

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -32,6 +32,8 @@ jobs:
     - name: Run tests
       run: |
         # Check that the expected output is printed
-        ./examples/array_multiplication/build/src/kompute_array_mult | \
-          tee $(tty) | grep -q "Output: {  0  4  12  }"
+        TEST_STDOUT=./examples/array_multiplication/build/src/kompute_array_mult
+        TEST_CODE=$?
+        TEST_MATCH= echo $TEST_STDOUT | grep -q "Output: {  0  4  12  }"
+        exit $(($TEST_CODE + $TEST_MATCH))
 

--- a/.github/workflows/cpp_examples.yml
+++ b/.github/workflows/cpp_examples.yml
@@ -32,7 +32,7 @@ jobs:
     - name: Run tests
       run: |
         # Check that the expected output is printed
-        OUTPUT=${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult | \
+        ${{github.workspace}}/examples/array_multiplication/build/src/kompute_array_mult | \
           tee $(tty) | grep -q "Output: {  0  4  12  }"
 
 

--- a/.github/workflows/cpp_tests.yml
+++ b/.github/workflows/cpp_tests.yml
@@ -10,9 +10,9 @@ jobs:
   cpp-tests-debug-with-debug-layers:
     runs-on: ubuntu-latest
     container: axsauze/kompute-builder:0.4
+    env:
+      VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
     steps:
-    - name: Install vulkaninfo
-      run: apt update -y && apt install -y vulkan-tools
     - name: Checkout
       uses: actions/checkout@v3
       with:
@@ -27,15 +27,15 @@ jobs:
         build-type: Debug
         run-test: false
         ctest-options: -V
-        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=OFF -DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
+        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=OFF -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
     - name: Run tests
-      env:
-        VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
       run: make mk_run_tests
-  
+
   cpp-tests-release-with-debug-layers:
     runs-on: ubuntu-latest
     container: axsauze/kompute-builder:0.4
+    env:
+      VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
     steps:
     - name: Checkout
       uses: actions/checkout@v3
@@ -51,15 +51,15 @@ jobs:
         build-type: Release
         run-test: false
         ctest-options: -V
-        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=OFF -DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
+        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=OFF -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
     - name: Run tests
-      env:
-        VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
       run: make mk_run_tests
 
   cpp-tests-debug-without-debug-layers:
     runs-on: ubuntu-latest
     container: axsauze/kompute-builder:0.4
+    env:
+      VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
     steps:
     - name: Checkout
       uses: actions/checkout@v3
@@ -75,15 +75,15 @@ jobs:
         build-type: Debug
         run-test: false
         ctest-options: -V
-        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=ON -DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
+        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=ON -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
     - name: Run tests
-      env:
-        VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
       run: make mk_run_tests
   
   cpp-tests-release-without-debug-layers:
     runs-on: ubuntu-latest
     container: axsauze/kompute-builder:0.4
+    env:
+      VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
     steps:
     - name: Checkout
       uses: actions/checkout@v3
@@ -99,8 +99,6 @@ jobs:
         build-type: Release
         run-test: false
         ctest-options: -V
-        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=ON -DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
+        configure-options: -DKOMPUTE_OPT_BUILD_TESTS=ON -DKOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS=ON -DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=ON
     - name: Run tests
-      env:
-        VK_ICD_FILENAMES: "/swiftshader/vk_swiftshader_icd.json"
       run: make mk_run_tests

--- a/examples/android/android-simple/app/src/main/cpp/CMakeLists.txt
+++ b/examples/android/android-simple/app/src/main/cpp/CMakeLists.txt
@@ -2,13 +2,22 @@ cmake_minimum_required(VERSION 3.20)
 
 set(CMAKE_CXX_STANDARD 17)
 
-include(FetchContent)
-FetchContent_Declare(kompute GIT_REPOSITORY https://github.com/KomputeProject/kompute.git
-                             GIT_TAG 1344ece4ac278f9b3be3b4555ffaace7a032b91f) # The commit hash for a dev version before v0.9.0. Replace with the latest from: https://github.com/KomputeProject/kompute/releases
+# Options
+option(KOMPUTE_OPT_GIT_TAG "The tag of the repo to use for the example" 1344ece4ac278f9b3be3b4555ffaace7a032b91f)
+option(KOMPUTE_OPT_FROM_SOURCE "Whether to build example from source or from git fetch repo" 0)
+
 set(KOMPUTE_OPT_ANDROID_BUILD ON)
 set(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS ON)
-FetchContent_MakeAvailable(kompute)
-include_directories(${kompute_SOURCE_DIR}/src/include)
+
+if(KOMPUTE_OPT_FROM_SOURCE)
+    add_subdirectory(../../../../../../../ ${CMAKE_CURRENT_BINARY_DIR}/kompute_build)
+else()
+    include(FetchContent)
+    FetchContent_Declare(kompute GIT_REPOSITORY https://github.com/KomputeProject/kompute.git
+        GIT_TAG ${KOMPUTE_OPT_GIT_TAG})
+    FetchContent_MakeAvailable(kompute)
+    include_directories(${kompute_SOURCE_DIR}/src/include)
+endif()
 
 # Add to the list, so CMake can later find the code to compile shaders to header files
 list(APPEND CMAKE_PREFIX_PATH "${kompute_SOURCE_DIR}/cmake")

--- a/examples/array_multiplication/CMakeLists.txt
+++ b/examples/array_multiplication/CMakeLists.txt
@@ -3,6 +3,10 @@ project(kompute_array_mult)
 
 set(CMAKE_CXX_STANDARD 14)
 
+# Options
+option(KOMPUTE_OPT_GIT_TAG "The tag of the repo to use for the example" 1344ece4ac278f9b3be3b4555ffaace7a032b91f)
+option(KOMPUTE_OPT_FROM_SOURCE "Whether to build example from source or from git fetch repo" 0)
+
 # Set a default build type if none was specified
 # Based on: https://github.com/openchemistry/tomviz/blob/master/cmake/BuildType.cmake
 set(DEFAULT_BUILD_TYPE "Release")
@@ -24,11 +28,15 @@ if(WIN32) # Install dlls in the same directory as the executable on Windows
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 endif()
 
-include(FetchContent)
-FetchContent_Declare(kompute GIT_REPOSITORY https://github.com/KomputeProject/kompute.git
-    GIT_TAG 1344ece4ac278f9b3be3b4555ffaace7a032b91f) # The commit hash for a dev version before v0.9.0. Replace with the latest from: https://github.com/KomputeProject/kompute/releases
-FetchContent_MakeAvailable(kompute)
-include_directories(${kompute_SOURCE_DIR}/src/include)
+if(KOMPUTE_OPT_FROM_SOURCE)
+    add_subdirectory(../../ ${CMAKE_CURRENT_BINARY_DIR}/kompute_build)
+else()
+    include(FetchContent)
+    FetchContent_Declare(kompute GIT_REPOSITORY https://github.com/KomputeProject/kompute.git
+        GIT_TAG ${KOMPUTE_OPT_GIT_TAG})
+    FetchContent_MakeAvailable(kompute)
+    include_directories(${kompute_SOURCE_DIR}/src/include)
+endif()
 
 # Add to the list, so CMake can later find the code to compile shaders to header files
 list(APPEND CMAKE_PREFIX_PATH "${kompute_SOURCE_DIR}/cmake")

--- a/examples/array_multiplication/src/main.cpp
+++ b/examples/array_multiplication/src/main.cpp
@@ -38,4 +38,8 @@ main()
         std::cout << elem << "  ";
     }
     std::cout << "}" << std::endl;
+
+    if (tensorOut->vector() != std::vector<float>{ 0, 4, 12 }) {
+        throw std::runtime_error("Result does not match");
+    }
 }

--- a/examples/logistic_regression/CMakeLists.txt
+++ b/examples/logistic_regression/CMakeLists.txt
@@ -3,6 +3,10 @@ project(kompute_logistic_regression)
 
 set(CMAKE_CXX_STANDARD 14)
 
+# Options
+option(KOMPUTE_OPT_GIT_TAG "The tag of the repo to use for the example" 1344ece4ac278f9b3be3b4555ffaace7a032b91f)
+option(KOMPUTE_OPT_FROM_SOURCE "Whether to build example from source or from git fetch repo" 0)
+
 # Set a default build type if none was specified
 # Based on: https://github.com/openchemistry/tomviz/blob/master/cmake/BuildType.cmake
 set(DEFAULT_BUILD_TYPE "Release")
@@ -24,11 +28,15 @@ if(WIN32) # Install dlls in the same directory as the executable on Windows
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 endif()
 
-include(FetchContent)
-FetchContent_Declare(kompute GIT_REPOSITORY https://github.com/KomputeProject/kompute.git
-    GIT_TAG 1344ece4ac278f9b3be3b4555ffaace7a032b91f) # The commit hash for a dev version before v0.9.0. Replace with the latest from: https://github.com/KomputeProject/kompute/releases
-FetchContent_MakeAvailable(kompute)
-include_directories(${kompute_SOURCE_DIR}/src/include)
+if(KOMPUTE_OPT_FROM_SOURCE)
+    add_subdirectory(../../ ${CMAKE_CURRENT_BINARY_DIR}/kompute_build)
+else()
+    include(FetchContent)
+    FetchContent_Declare(kompute GIT_REPOSITORY https://github.com/KomputeProject/kompute.git
+        GIT_TAG ${KOMPUTE_OPT_GIT_TAG})
+    FetchContent_MakeAvailable(kompute)
+    include_directories(${kompute_SOURCE_DIR}/src/include)
+endif()
 
 # Add to the list, so CMake can later find the code to compile shaders to header files
 list(APPEND CMAKE_PREFIX_PATH "${kompute_SOURCE_DIR}/cmake")

--- a/examples/logistic_regression/shader/my_shader.comp
+++ b/examples/logistic_regression/shader/my_shader.comp
@@ -1,6 +1,6 @@
 #version 450
 
-layout (constant_id = 0) const uint M = 0;
+layout (constant_id = 0) const float m = 0;
 
 layout (local_size_x = 1) in;
 
@@ -13,8 +13,6 @@ layout(set = 0, binding = 5) buffer bwoutj { float woutj[]; };
 layout(set = 0, binding = 6) buffer bbin { float bin[]; };
 layout(set = 0, binding = 7) buffer bbout { float bout[]; };
 layout(set = 0, binding = 8) buffer blout { float lout[]; };
-
-float m = float(M);
 
 float sigmoid(float z) {
     return 1.0 / (1.0 + exp(-z));
@@ -52,3 +50,4 @@ void main() {
 
     lout[idx] = calculateLoss(yHat, yCurr);
 }
+

--- a/examples/logistic_regression/src/main.cpp
+++ b/examples/logistic_regression/src/main.cpp
@@ -63,4 +63,10 @@ main()
     std::cout << "w1: " << wIn->data()[0] << std::endl;
     std::cout << "w2: " << wIn->data()[1] << std::endl;
     std::cout << "b: " << bIn->data()[0] << std::endl;
+
+    if (wIn->data()[0] > 0.0 ||
+            wIn->data()[1] < 1.0 ||
+            bIn->data()[0] > 0.0) {
+        throw std::runtime_error("Result does not match");
+    }
 }

--- a/examples/logistic_regression/src/main.cpp
+++ b/examples/logistic_regression/src/main.cpp
@@ -21,35 +21,41 @@ main()
     std::shared_ptr<kp::TensorT<float>> y = mgr.tensor({ 0, 0, 0, 1, 1 });
 
     std::shared_ptr<kp::TensorT<float>> wIn = mgr.tensor({ 0.001, 0.001 });
-    std::shared_ptr<kp::TensorT<float>> wOutI = mgr.tensor({ 0, 0, 0, 0, 0 });
-    std::shared_ptr<kp::TensorT<float>> wOutJ = mgr.tensor({ 0, 0, 0, 0, 0 });
+    std::shared_ptr<kp::TensorT<float>> wOutI =
+      mgr.tensor({ 0, 0, 0, 0, 0 });
+    std::shared_ptr<kp::TensorT<float>> wOutJ =
+      mgr.tensor({ 0, 0, 0, 0, 0 });
 
     std::shared_ptr<kp::TensorT<float>> bIn = mgr.tensor({ 0 });
-    std::shared_ptr<kp::TensorT<float>> bOut = mgr.tensor({ 0, 0, 0, 0, 0 });
+    std::shared_ptr<kp::TensorT<float>> bOut =
+      mgr.tensor({ 0, 0, 0, 0, 0 });
 
-    std::shared_ptr<kp::TensorT<float>> lOut = mgr.tensor({ 0, 0, 0, 0, 0 });
+    std::shared_ptr<kp::TensorT<float>> lOut =
+      mgr.tensor({ 0, 0, 0, 0, 0 });
 
-    const std::vector<std::shared_ptr<kp::Tensor>> params = {
-        xI, xJ, y, wIn, wOutI, wOutJ, bIn, bOut, lOut
-    };
-
-    const std::vector<uint32_t> shader = std::vector<uint32_t>(
-      shader::MY_SHADER_COMP_SPV.begin(), shader::MY_SHADER_COMP_SPV.end());
-
-    std::shared_ptr<kp::Algorithm> algo = mgr.algorithm(
-      params, shader, kp::Workgroup({ 5 }), std::vector<float>({ 5.0 }));
+    std::vector<std::shared_ptr<kp::Tensor>> params = { xI,  xJ,    y,
+                                                        wIn, wOutI, wOutJ,
+                                                        bIn, bOut,  lOut };
 
     mgr.sequence()->eval<kp::OpTensorSyncDevice>(params);
+
+    std::vector<uint32_t> spirv2{ 0x1, 0x2 };
+
+    std::vector<uint32_t> spirv(
+      shader::MY_SHADER_COMP_SPV.begin(),
+      shader::MY_SHADER_COMP_SPV.end());
+
+    std::shared_ptr<kp::Algorithm> algorithm = mgr.algorithm(
+      params, spirv, kp::Workgroup({ 5 }), std::vector<float>({ 5.0 }));
 
     std::shared_ptr<kp::Sequence> sq =
       mgr.sequence()
         ->record<kp::OpTensorSyncDevice>({ wIn, bIn })
-        ->record<kp::OpAlgoDispatch>(algo)
+        ->record<kp::OpAlgoDispatch>(algorithm)
         ->record<kp::OpTensorSyncLocal>({ wOutI, wOutJ, bOut, lOut });
 
     // Iterate across all expected iterations
     for (size_t i = 0; i < ITERATIONS; i++) {
-
         sq->eval();
 
         for (size_t j = 0; j < bOut->size(); j++) {
@@ -59,12 +65,12 @@ main()
         }
     }
 
-    std::cout << "RESULTS" << std::endl;
-    std::cout << "w1: " << wIn->data()[0] << std::endl;
-    std::cout << "w2: " << wIn->data()[1] << std::endl;
-    std::cout << "b: " << bIn->data()[0] << std::endl;
+    KP_LOG_WARN("Result wIn i: {}, wIn j: {}, bIn: {}",
+                wIn->data()[0],
+                wIn->data()[1],
+                bIn->data()[0]);
 
-    if (wIn->data()[0] > 0.0 ||
+    if (wIn->data()[0] > 0.01 ||
             wIn->data()[1] < 1.0 ||
             bIn->data()[0] > 0.0) {
         throw std::runtime_error("Result does not match");

